### DISCUSSION TITLE
Is there a way to `@Inject` an `Emitter` without `@Channel`

### DISCUSSION THREAD

**Discussion by hknlof** [smallrye/smallrye-reactive-messaging#2053] [General]
`@Channel` is annotated by `@Qualifier`, which in turn leads to the class, that uses `@Channel` to be registered as a bean.

Is there a way to inject or instantiate an `Emitter` without registering the using class as a bean? Constructor injection comes to my mind, but then the question how to provide the emitter with the correct stream name.

Refs:
- https://smallrye.io/smallrye-reactive-messaging/3.22.1/concepts/emitter/
- https://github.com/smallrye/smallrye-reactive-messaging/blob/main/api/src/main/java/io/smallrye/reactive/messaging/annotations/Channel.java#L45

**Comment by cescoffier** [comment]
I don't think so. An emitter is a managed object. So it is expected to be injected into a bean.

You can try to use the CDI programmatic API but you will need @Channel (but as a literal)

**Reply by hknlof** [reply]
Thanks for clarification. Will then have to use a proxy class for producing my desired beans.
